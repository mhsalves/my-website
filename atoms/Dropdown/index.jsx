import React from 'react';
import {
  node, func, bool,
} from 'prop-types';

import {
  DropdownContainer,
  contentStyle,
} from './styles';

function Dropdown(props) {
  const { handler, content, isOpen } = props;

  return (
    <DropdownContainer>
      {handler}
      {content({ contentStyle, isOpen })}
    </DropdownContainer>
  );
}

Dropdown.propTypes = {
  handler: node.isRequired,
  content: func.isRequired,
  isOpen: bool,
};

Dropdown.defaultProps = {
  isOpen: false,
};

export default Dropdown;
