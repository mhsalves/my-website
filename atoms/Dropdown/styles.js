import styled, { css } from 'styled-components';

const showStyle = css`
  opacity: ${({ isOpen }) => (isOpen ? '1' : '0')};
  visibility: ${({ isOpen }) => (isOpen ? 'visible' : 'hidden')};
`;

const contentStyle = css`
  position: absolute;
  top: 100%;
  right: 0;
  left: 0;

  transition: opacity .2s ease, visibility .2s ease;
  ${showStyle}
`;

const DropdownContainer = styled.div`
  position: relative;
  display: inline;
`;

export {
  DropdownContainer,
  contentStyle,
};
