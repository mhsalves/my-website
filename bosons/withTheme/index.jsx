import React from 'react';
import { ThemeProvider } from 'styled-components';
import baseTheme from '../themes/base';

const withTheme = Component => props => (
  <ThemeProvider theme={baseTheme}>
    <Component {...props} />
  </ThemeProvider>
);

export default withTheme;
