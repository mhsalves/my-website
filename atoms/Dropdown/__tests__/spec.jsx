import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';

import Dropdown from '../index';

import withTheme from '../../../bosons/withTheme';

const ComponentWithTheme = withTheme(Dropdown);
const Content = styled.div`${({ contentStyle }) => contentStyle}`;

const handler = (<button type="button">Click here</button>);
const content = contentProps => (<Content {...contentProps}>Hello World !</Content>);

describe('given: Dropdown', () => {
  test('when: is not opened', () => {
    const component = renderer.create(
      <ComponentWithTheme handler={handler} content={content} />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  test('when: is opened', () => {
    const component = renderer.create(
      <ComponentWithTheme handler={handler} content={content} isOpen />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
