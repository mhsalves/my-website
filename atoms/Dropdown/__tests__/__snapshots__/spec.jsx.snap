// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given: Dropdown when: is not opened 1`] = `
.c0 {
  position: relative;
  display: inline;
}

.c1 {
  position: absolute;
  top: 100%;
  right: 0;
  left: 0;
  -webkit-transition: opacity .2s ease,visibility .2s ease;
  transition: opacity .2s ease,visibility .2s ease;
  opacity: 0;
  visibility: hidden;
}

<div
  className="c0"
>
  <button
    type="button"
  >
    Click here
  </button>
  <div
    className="c1"
  >
    Hello World !
  </div>
</div>
`;

exports[`given: Dropdown when: is opened 1`] = `
.c0 {
  position: relative;
  display: inline;
}

.c1 {
  position: absolute;
  top: 100%;
  right: 0;
  left: 0;
  -webkit-transition: opacity .2s ease,visibility .2s ease;
  transition: opacity .2s ease,visibility .2s ease;
  opacity: 1;
  visibility: visible;
}

<div
  className="c0"
>
  <button
    type="button"
  >
    Click here
  </button>
  <div
    className="c1"
  >
    Hello World !
  </div>
</div>
`;
